scripts/ccf/sandbox_local/scale-nodes.sh

```diff
@@ -0,0 +1,140 @@
+}
+
+self() {
+    jq -c '.[]' | while read -r node; do
+        url=$(echo $node | node-url | tr -d '"')
+        curl -s https://$url/node/network/nodes/self \
+            --cacert $KMS_SERVICE_CERT_PATH
+    done | jq -s '.'
+}
+
+container-state-is() {
+    jq -c '.[]' | while read -r node; do
+        url=$(echo $node | node-url | tr -d '"')
+        resp=$(curl -s https://$url/node/network/nodes/self \
+            --cacert $KMS_SERVICE_CERT_PATH \
+            -o /dev/null -w "%{http_code}")
+        if [[ $1 = "Running" ]] && [ "$resp" -eq 200 ]; then
+            echo "$node"
+        elif [[ $1 = "Stopped" ]] && [ "$resp" -ne 200 ]; then
+            echo "$node"
+        fi
+    done | jq -s '.'
+}
+
+running() {
+    status-is "Trusted" | container-state-is "Running"
+}
+
+count() {
+    jq 'length'
+}
+
+iterate() {
+    jq -c '.[]' | tr -d '"'
+}
+
+ccf-sandbox-local-scale-nodes() {
+
+    REPO_ROOT="$(realpath "$(dirname "$(realpath "${BASH_SOURCE[0]}")")/../../..")"
+    source $REPO_ROOT/scripts/ccf/propose.sh
+
+    node_count=""
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+            -n|--node-count)
+                node_count="$2"
+                shift 2
+                ;;
+            *)
+                echo "Unknown parameter: $1"
+                return 1
+                ;;
+        esac
+    done
+
+    if [ -z $node_count ]; then
+        read -p "Enter node count to scale to: " node_count
+    fi
+
+    current_node_count=$(ccf-sandbox-local-nodes | running | count)
+    node_count_difference=$((node_count - current_node_count))
+
+    # Scale the CCF node containers
+    docker compose \
+        -f $REPO_ROOT/services/docker-compose.yml \
+        up ccf-sandbox-joiner \
+            --scale ccf-sandbox-joiner=$((node_count - 1)) \
+            --wait
+
+    # Propose adding any new nodes
+    ccf-sandbox-local-nodes | status-is "Pending" | node-id | iterate \
+        | while read -r node_id; do
+        proposal_file="$WORKSPACE/proposals/trust_${node_id}.json"
+        NODE_ID=$node_id \
+        VALID_FROM=$(date -u +"%y%m%d%H%M%SZ") \
+            envsubst < "$REPO_ROOT/governance/proposals/transition_node_to_trusted.json" \
+                | jq > "$proposal_file"
+        ccf-propose "$proposal_file"
+    done
+
+    # Propose removing any stopped nodes
+    if [ "$node_count_difference" -lt 0 ]; then
+        ccf-sandbox-local-nodes | status-is "Trusted" | container-state-is "Stopped" \
+            | node-id | iterate | while read -r node_id; do
+            proposal_file="$WORKSPACE/proposals/remove_${node_id}.json"
+            NODE_ID=$node_id \
+                envsubst < "$REPO_ROOT/governance/proposals/remove_node.json" \
+                    | jq > "$proposal_file"
+            ccf-propose "$proposal_file"
+        done
+    fi
+
+    # Wait until the node count matches the desired value
+    retries=10
```

Comment:
will 10 secs be deterministic? I assume it will depend on the number of nodes.

Reply:
It's a good question, and thinking about it I realised for non auto-accept resolve function constitutions, this will always fail since the trust proposal won't be accepted until it's voted on.

So in https://github.com/microsoft/azure-privacy-sandbox-kms/pull/259/commits/38892485ef3cb488badfd8635bbe0723c72cb488 I replace this with a loop which will print outstanding proposals so the users can then easily vote on them